api/graphql/types/opening_hours/types.py

```diff
@@ -1,187 +1,58 @@
 from django.utils import timezone
 
-from opening_hours.enums import State as ResourceState
-from opening_hours.utils.opening_hours_client import OpeningHoursClient
+from common.typing import GQLInfo
 from reservation_units.models import ReservationUnit
 from spaces.models import Unit
 
 DEFAULT_TIMEZONE = timezone.get_default_timezone()
 
 
-def get_time_as_utc(time: datetime.time, date: datetime.date, tz_info: timezone):
-    """
-    Gets utc time from given time, date and timezone information.
-    Graphene seems to render time with offset info (e.g. 10:00:00+00:00) for localized to UTC only.
-    So with e.g. time zone being 'Europe/Helsinki' just don't work for some reason not yet known.
-    So for now we render all as UTC using this little helper.
-    """
-    start_dt = datetime.datetime.combine(date, time)
-    start_dt = start_dt.astimezone(tz_info)
-    start_dt = start_dt.astimezone(timezone.utc)
-    return start_dt.timetz()
-
-
-class TimeSpanType(graphene.ObjectType):
-    start_time = graphene.Time()
-    end_time = graphene.Time()
-    weekdays = graphene.List(graphene.Int)
-    resource_state = graphene.String()
-    end_time_on_next_day = graphene.Boolean()
-    name_fi = graphene.String()
-    name_en = graphene.String()
-    name_sv = graphene.String()
-    description_fi = graphene.String()
-    description_en = graphene.String()
-    description_sv = graphene.String()
-
-    def resolve_start_time(self, info):
-        if not self.start_time:
-            return None
-        tzinfo = self.start_time.tzinfo or DEFAULT_TIMEZONE
-        start = get_time_as_utc(self.start_time, datetime.date.today(), tzinfo)
-        return start
-
-    def resolve_end_time(self, info):
-        if not self.end_time:
-            return None
-        tzinfo = self.start_time.tzinfo or DEFAULT_TIMEZONE
-        end = get_time_as_utc(self.end_time, datetime.date.today(), tzinfo)
-        return end
+class ReservableTimeSpanType(graphene.ObjectType):
+    start_datetime = graphene.DateTime()
+    end_datetime = graphene.DateTime()
 
+    def resolve_start_datetime(self, info: GQLInfo):
+        return self.start_datetime
 
-class PeriodType(graphene.ObjectType):
-    period_id = graphene.Int()
-    start_date = graphene.Date()
-    end_date = graphene.Date()
-    resource_state = graphene.String()
-    time_spans = graphene.List(TimeSpanType)
-    name_fi = graphene.String()
-    name_en = graphene.String()
-    name_sv = graphene.String()
-    description_fi = graphene.String()
-    description_en = graphene.String()
-    description_sv = graphene.String()
+    def resolve_end_datetime(self, info: GQLInfo):
+        return self.end_datetime
 
 
-class OpeningTimesType(graphene.ObjectType):
-    date = graphene.Date()
-    start_time = graphene.DateTime()
-    end_time = graphene.DateTime()
-    state = graphene.String()
-    periods = graphene.List(graphene.Int)
-    is_reservable = graphene.Boolean()
+class ReservableTimeSpansGraphQLMixin:
```

Comment:
Seems like this mixin is only used once (by `api.graphql.types.reservation_units.types.ReservationUnitByPkType`). Should just merge into that class.